x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -314,6 +298,48 @@ public static TemporalAmount parseTemporalAmount(Object val, DataType expectedTy
         throw new ParsingException(Source.EMPTY, errorMessage, val, expectedType);
     }
 
+    public static TemporalAmount maybeParseTemporalAmount(String str) {
+        // The string literal can be either Date_Period or Time_Duration, derive the data type from its qualifier
+        String errorMessage = "Cannot parse [{}] to {}";
+        String expectedTypes = DATE_PERIOD + " or " + TIME_DURATION;
+        StringBuilder value = new StringBuilder();
+        StringBuilder qualifier = new StringBuilder();
```

Comment:
I find `qualifier` as a misnomer, if I understood what this is. If this is a **time unit** or **unit of time** (days, seconds, months etc) I find it more clear named as `timeUnit`. And the same goes for the error messages. `Cannot convert string [    1    fox   ] to [DATE_PERIOD or TIME_DURATION], error [Unexpected time interval qualifier: 'fox']"`

Reply:
Yeah, `qualifier` is not the best name, `TimeUnit` does look better. I search and found that `TemporalUnit` is used by JDK, I'll use `TemporalUnit`, also change the error messages.